### ISSUE TITLE
Version 0.0.5 can not play video (windows flashes and close unexpectedly)

### ISSUE THREAD

**Issue by wangdsh** [lhc70000/iina#377]
**System and IINA version:**

- macOS 10.11.6
- IINA 0.0.5

can not open mp4 file
can not play flv file, no picture and sound

I'd better use version 0.0.4 which is good on my computer

**Comment by waxfingers** [comment]
OSX 110.11.6 here, iina's working perfectly both with mp4 and flv. Maybe's something related with your system (?)

**Comment by lhc70000** [comment]
It should be the same problem with #339, #322, #311, and 0.0.4 also have this bug. It's not related to particular files or file types, just IINA itself.

Please quit IINA completely and try open again when encountered this problem, and we'll try to fix it as soon as possible.

**Comment by wangdsh** [comment]
Now it's OK! Maybe I wasn't uninstall IINA completely. Sorry about my issue.

**Comment by lhc70000** [comment]
Reopen to be easily searched by users and reduce duplicate issues

**Comment by lhc70000** [comment]
Let me list my observations:

I reproduced this bug in a very slow virtual machine for once, which enabled me to have a detailed look on it:

- It showed **TWO** window on launch.
- One of them (the one showed earlier) disappeared immediately
- The other started loading file
- mpv failed to play the file (according to log, the reason was "no video", see `5.223`)
- It tried to play following items in the playlist, failed to
- Finished playing, window closed.
- If user tries to open file again without quitting IINA, it will say `OpenGL context not initialized` (`9.995` in log), which should be done in `windowDidLoad`, and it was initialized in `4.972` according to log.

log: [log_for_377.txt](https://github.com/lhc70000/iina/files/776538/log_for_377.txt)

Other facts:

- Seems that it only happens when user open IINA after long time, or after a restart; but I can't reproduce it by restarting system.
- It really shows two windows on launch in a very low possibility, one of which is not initialized (like in IB), and the other plays the video, but no video displayed (black window).
- `MainWindowController.deinit` is called on launch when this bug happens.
- mpv properties like playlist item type (must be one of `audio`, `video` and `sub`) return empty string when this bug happens.
- Tried disabling "visible on launch" and window restoration in xib, didn't work.

My guess:

For some reason, a (cached?) window links to `playerCore` and other components, but is closed and replaced by a new one.


@saagarjha Do you have any thoughts on this?

**Comment by lhc70000** [comment]
Some log that may related to the deinit of one window:

```
12/2/17 8:19:32.978 pm IINA[340]: *** Assertion failure in -[NSWindow _changeJustMain], /SourceCache/AppKit/AppKit-1348.17/AppKit.subproj/NSWindow.m:14319
12/2/17 8:19:32.978 pm IINA[340]: Invalid parameter not satisfying: [self canBecomeMainWindow]
```

**Comment by saagarjha** [comment]
I have not encountered this issue; would you mind sending me a list of steps to reproduce it?

**Comment by lhc70000** [comment]
@saagarjha 

Sadly, that's the bottleneck: it's very hard for me to reproduce this issue (only 3~4 times in last two weeks), but ~10 users reported the same problem and some of them said that it happens almost every time when restarted system. For me, it seems that only happens when opening IINA after long time (works well even after restart).

Seems that it's a bug from 0.0.1 (in 0.0.1 it crashed at `WindowController.deinit`). Do you think there're any problem or bad practice with creating/managing WindowController in current code; or any reason Cocoa deiniting a window on launch?

**Comment by lhc70000** [comment]
Fixed in 0.0.6.
